src/doc/en/developer/coding_in_python.rst

```diff
@@ -38,14 +38,36 @@ using one of two mechanisms:
 - Backport packages
 
   - `importlib_metadata <../reference/spkg/importlib_metadata>`_
-    (to be used in place of ``importlib.metadata``),
+    (to be used in place of ``importlib.metadata`` when Python older
+    than 3.11 is in use if you need a newer feature),
   - `importlib_resources <../reference/spkg/importlib_resources>`_
-    (to be used in place of ``importlib.resources``),
+    (to be used in place of ``importlib.resources`` when Python older than
+    3.11 is in use if you need a newer feature),
   - `typing_extensions <../reference/spkg/typing_extensions>`_
-    (to be used in place of ``typing``).
+    (to be used in place of ``typing`` when Python older than
+    3.11 is in use and any of the features introduced after Python
+    3.9.0 are used).
 
-  The Sage library declares these packages as dependencies and ensures that
-  versions that provide features of Python 3.11 are available.
+  The Sage library declares these backport packages as conditional
+  dependencies for use with Python older than 3.11 and ensures that
+  versions of the packages that provide features of Python 3.11
+  are available.
+
+  The backport packages should not be imported unconditionally.
+  For example, on Linux distributions that only ship newer versions
+  of Python, the backport packages may not be available as
+  system packages. Hence, if there is a need to use the backport
+  packages, they should be imported like this:
+
+  .. CODE-BLOCK:: python
+
+        import sys
+
+        if sys.version_info < (3, 11):
+            # Use backport package providing Python 3.11 features
+            from importlib_resources import files
+        else:
+            from importlib.resources import files
```

Comment:
This gives the impression that we recommend using the backport packages. IMO, we should discourage the backport packages. Right now, afaict, nothing in the sage library requires the backport packages and we should keep it that way.

The risk is that if someone uses the backport packages it might use a newer feature not available on current python. We've already pointed out that distros are not shipping some of the backport packages.

Please remove this suggestion.

Reply:
It's documenting the status quo, which was established by positively reviewed PRs in the past.